media-libs/jbigkit/jbigkit-2.1.recipe

```diff
@@ -0,0 +1,88 @@
+SUMMARY="A lossless image compression library"
+DESCRIPTION="JBIG-KIT is a software implementation of the JBIG1 data \
+compression standard (ITU-T T.82), which was designed for bi-level image \
+data, such as scanned documents. This library is available in portable C \
+code. It is widely used in fax products, printer firmware, printer drivers, \
+document management systems and imaging software."
+HOMEPAGE="https://www.cl.cam.ac.uk/~mgk25/jbigkit/"
+COPYRIGHT="1995-2014 Markus Kuhn
+	1989, 1991 Jef Poskanzer"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://www.cl.cam.ac.uk/~mgk25/download/jbigkit-$portVersion.tar.gz"
+CHECKSUM_SHA256="de7106b6bfaf495d6865c7dd7ac6ca1381bd12e0d81405ea81e7f2167263d932"
+PATCHES="jbigkit-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 ?x86_64"
+
+libVersion=$portVersion
+libVersionCompat="$libVersion compat >= 2"
+
+PROVIDES="
+	jbigkit = $portVersion
+	cmd:jbgtopbm
+	cmd:jbgtopbm85
+	cmd:pbmtojbg
+	cmd:pbmtojbg85
+	lib:libjbig = $libVersionCompat
+	lib:libjbig85 = $libVersionCompat
+	"
+REQUIRES="
+	haiku
+	"
+
+PROVIDES_devel="
+	jbigkit_devel = $portVersion
+	devel:libjbig = $libVersionCompat
+	devel:libjbig85 = $libVersionCompat
+	"
+REQUIRES_devel="
+	jbigkit == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc
+	cmd:make
+	"
+defineDebugInfoPackage jbigkit \
+	$libDir/libjbig.so \
+	$libDir/libjbig85.so
```

Comment:
Shouldn't this be done on the main file instead of the symlink? And aren't both `libjbig` and `libjbig85` the same file?

Reply:
Have to check this one ...

Reply:
I think both work, but it's better, imo, to specify the file instead of the symlink.

```diff
 defineDebugInfoPackage jbigkit \
-	$libDir/libjbig.so \
+	"$libDir"/libjbig.so.$libVersion \
-	$libDir/libjbig85.so
+	"$libDir"/libjbig85.so.$libVersion
```
The main advantage is that if we forget to update the `libVersion` then haikuporter will complain, so we have a chance to notice it before merging.

BTW, if I'm not mistaken, we should add support for secondary architectures because this recipe provides a lib. But maybe there are exceptions.

Reply:
Well, we should add it anyway as lots of progs out there won't compile with gcc2.

Reply:
This doesn't seem to work?

```
objcopy: '/packages/jbigkit_x86-2.1-1/.self/lib/x86/libjbig.so.2.1': No such file
```

Reply:
Still a no go: 

```
objcopy: error: the input file '/packages/jbigkit-2.1-1/.self/lib/libjbig.so.2.1' is empty
```